code-studio/src/js/pd/workshop_dashboard/attendance/session_attendance_row.jsx

```diff
@@ -0,0 +1,91 @@
+/* global React */
+
+var OverlayTrigger = require('react-bootstrap').OverlayTrigger;
+var Tooltip = require('react-bootstrap').Tooltip;
+
+var SessionAttendanceRow = React.createClass({
+  propTypes: {
+    sessionId: React.PropTypes.number,
+    attendance: React.PropTypes.shape({
+      name: React.PropTypes.string.isRequired,
+      email: React.PropTypes.string.isRequired,
+      enrolled: React.PropTypes.bool.isRequired,
+      user_id: React.PropTypes.number,
+      in_section: React.PropTypes.bool.isRequired,
+      attended: React.PropTypes.bool.isRequired
+    }).isRequired,
+    adminOverride: React.PropTypes.bool,
+    onChange: React.PropTypes.func.isRequired,
+    isReadOnly: React.PropTypes.bool
+  },
+
+  isValid: function () {
+    // Must have an account and have joined the section before being marked attended,
+    // unless overridden by an admin.
+    return (this.props.attendance.user_id && this.props.attendance.in_section) || this.props.adminOverride;
+  },
+
+  handleClickAttended: function () {
+    if (this.isValid()) {
+      this.props.onChange();
+    }
+  },
+
+  renderAttendedCellContents: function () {
+    var checkBoxClass = this.props.attendance.attended ? "fa fa-check-square-o" : "fa fa-square-o";
+    if (this.props.isReadOnly) {
+      return (
+        <div>
+          <i className={checkBoxClass}/>
+        </div>
+      );
+    }
+
+    var contents = (
+      <div style={{height: '100%', width: '100%', cursor:'pointer'}} onClick={this.handleClickAttended}>
+        <i className={checkBoxClass}/>
+      </div>
+    );
+
+    if (!this.isValid()) {
+      var tooltip = (
+        <Tooltip id={0}>
+          Teachers must have a Code Studio account and join the section before they can be marked attended.
+        </Tooltip>
+      );
+      return (
+        <OverlayTrigger overlay={tooltip} placement="left" delayShow={500}>
+          {contents}
+        </OverlayTrigger>
+      );
+    }
+
+    return contents;
+  },
```

Comment:
Another option instead of having a renderAttendedCellContents method would be to just have an AttendedCellContents component. You could do this with a functional stateless component, which becomes super easy with ES6, i.e. something like:
```jsx
const StringInADiv = props => (
  <div>
    {props.string}
  </div>
);
```